src/ui/src/components/core/base/BaseDropdown.vue

```diff
@@ -0,0 +1,133 @@
+<script setup lang="ts">
+import { onMounted, onUnmounted, PropType, ref } from "vue";
+
+defineProps({
+	options: {
+		type: Object as PropType<Record<string, string>>,
+		required: true,
+	},
+});
+
+const emits = defineEmits({
+	selected: (key: string) => typeof key === "string",
+});
+
+/**
+ * @deprecated use `useId` from Vue 3.5 when we upgrade
+ */
+function useId() {
+	return Math.random().toString();
+}
+
+const trigger = ref<Element>();
+
+const popoverId = useId();
+const popover = ref<Element & { hidePopover: () => void }>();
+const closePopover = () => popover.value?.hidePopover();
+
+const popoverTop = ref("unset");
+const popoverLeft = ref("unset");
+
+function computePopoverPosition() {
+	// the popover use an overlay with an absolute position, so we have to positionate it manually according to the trigger position
+	const boundingRect = trigger.value.getBoundingClientRect();
+	popoverTop.value = `${boundingRect.top + boundingRect.height + 4 + window.pageYOffset}px`;
+	popoverLeft.value = `${boundingRect.left + window.pageXOffset}px`;
+}
+
+function onItemClick(key: string) {
+	emits("selected", key);
+	closePopover();
+}
+
+onMounted(() => window.addEventListener("scroll", closePopover));
+onUnmounted(() => window.removeEventListener("scroll", closePopover));
+</script>
+
+<template>
```

Comment:
In all our component, the `template` go first, then we have `script setup`, then `script` then style. It's different here. I would prefer to stick to the existing convention if we can.